test/persist_newapi_tests.js

```diff
@@ -194,6 +194,14 @@ describe('persistor transaction checks', function () {
         });
     });
 
+    it('persistorFetchById with fetch spec with type projections', function () {
+        return Employee.persistorFetchById(empId, {fetch: { homeAddress: {fetch: {phone: true}}, roles: true}, projection: { Address: ['city'], Role: ['name'], Phone: ['']}}).then(function(employee) {
+            expect(employee.homeAddress.state).is.equal(undefined);
```

Comment:
revisit this for fetching all persistor related properties, foreign keys but not primitives. Maybe should be `Phone: null` or `Phone: []`